### PR TITLE
Updates GET articles method to allow for topic query and updates endpoints file

### PR DESCRIPTION
**Pull request by darinaJur**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -329,4 +329,53 @@ describe("GET /api/users", () => {
...
+        .get("/api/articles?topic=cats")
+        .expect(200)
+        .then((res) => {
+            res.body.articles.forEach((article) => {
+                expect(article).toMatchObject({
+                    author: expect.any(String),
+                    title: expect.any(String),
+                    article_id: expect.any(Number),
+                    // body: expect.any(String),
+                    topic: "cats",
```
Comment (nc-aimee-muter) [inline __tests__/app.test.js:346]: Nice

File: __tests__/app.test.js [hunk 2]
```diff
@@ -329,4 +329,53 @@ describe("GET /api/users", () => {
...
+        return request(app)
+        .get("/api/articles?topic=cats")
+        .expect(200)
+        .then((res) => {
+            res.body.articles.forEach((article) => {
+                expect(article).toMatchObject({
+                    author: expect.any(String),
+                    title: expect.any(String),
+                    article_id: expect.any(Number),
+                    // body: expect.any(String),
```
Comment (nc-aimee-muter) [inline __tests__/app.test.js:345]: Best to remove comments

File: __tests__/app.test.js [hunk 3]
```diff
@@ -329,4 +329,53 @@ describe("GET /api/users", () => {
             })
         })
     })
+})
+
+describe("GET /api/articles?topic", () => {
+    test("status 200: responds with articles of specified topic and if topic query is omitted, responds with all articles", () => {
```
Comment (nc-aimee-muter) [inline __tests__/app.test.js:335]: You could check the length too?

File: __tests__/app.test.js [hunk 4]
```diff
@@ -329,4 +329,53 @@ describe("GET /api/users", () => {
...
+    })
+    test("status 400: responds with 'Invalid Input' if the topic does not exist", () => {
+        return request(app)
+        .get("/api/articles?topic=dogs")
+        .expect(400)
+        .then((res) => {
+            expect(res.body.msg).toBe('Invalid Input')
+        })
+    })
+    test("status 200: responds with all articles when topic query is omitted", () => {
```
Comment (nc-aimee-muter) [inline __tests__/app.test.js:362]: You have already tested this previously so you don't need to repeat this test

File: __tests__/app.test.js [hunk 5]
```diff
@@ -329,4 +329,53 @@ describe("GET /api/users", () => {
...
+                    // body: expect.any(String),
+                    topic: "cats",
+                    created_at: expect.any(String),
+                    votes: expect.any(Number),
+                    article_img_url: expect.any(String)
+                })
+            })
+        })
+    })
+    test("status 400: responds with 'Invalid Input' if the topic does not exist", () => {
```
Comment (nc-aimee-muter) [inline __tests__/app.test.js:354]: This topic could exist at some point so it's not invalid, this would be a 404 and a better error message would be something like "No topic matching the topic query"

File: __tests__/app.test.js (hunk 3 above)
Comment (nc-aimee-muter) [inline __tests__/app.test.js:335]: Please update the 400 test to a 404 and add another 200 test to check that we get back an empty array when an existing topic has no associated articles.

File: models/articles.models.js [hunk 6]
```diff
@@ -1,72 +1,103 @@
...
+      if (rows.length === 0) {
+        return Promise.reject({ status: 404, msg: "Article not found" });
+      } else {
+        return rows[0];
+      }
+    });
+};
+
+exports.selectArticles = (topic) => {
+  const validTopic = ["cats", "mitch"];
```
Comment (nc-aimee-muter) [inline models/articles.models.js:20]: New topics could be added so your code would need to be more dynamic than this. Could you write a model function that checks for the existence of a topic?

**Review by nc-aimee-muter — COMMENTED** [review]
Some good testing a code here, one of the tests needs to be modified and there is a missing test also. Please make your code more dynamic (see comments)
